Source/Core/CesiumTerrainProvider.js

```diff
@@ -417,11 +425,19 @@ define([
 
         var skirtHeight = provider.getLevelMaximumGeometricError(level) * 5.0;
 
+        var rectangle = provider._tilingScheme.tileXYToRectangle(x, y, level);
+        var boundingOBB;
+        if (rectangle.width < CesiumMath.PI_OVER_TWO + 0.001) {
```

Comment:
I left out the `height` check because I think it will handle any height (up to pi) decently well.